test/test_roscpp/test/src/service_call.cpp

```diff
@@ -201,18 +201,24 @@ TEST(SrvCall, callSrvLongRunningTimeout)
   start = ros::SteadyTime::now();
   success = ros::service::call(service_name, req, res, short_timeout);
   d = ros::SteadyTime::now() - start;
+  printf("call with timeout of %f s took %f s\n", short_timeout.toSec(), d.toSec());
   ASSERT_FALSE(success)
       << service_name << " did NOT fail after timeout (res = " << res.str << ")";
   ASSERT_TRUE(d.toSec() < service_time_seconds)
```

Comment:
Use `ASSERT_LT(d.toSec(), service_time_seconds)` here and something similar below.
That way you can simplify the custom message on error, since gtest will print the expected and actual value for us.